python/social_bot/envs/grocery_ground.py

```diff
@@ -434,6 +435,9 @@ def __init__(self,
                  image_with_internal_states=False,
                  task_name='goal',
                  agent_type='pioneer2dx_noplugin',
+                 random_goal=None,
+                 fail_distance_thresh=3,
+                 max_steps=200,
```

Comment:
all of these can be configured through GroceryGroundGoalTask

Reply:
Oh, somehow missed that.

Reply:
Actually these are fixed around line 497 when initializing the GroceryGroundGoalTask.  This change makes it configurable.

Reply:
I also think it is better to configure through gin of GroceryGroundGoalTask.  GroceryGround is meant for many task, some of them does not need random_goal or fail_distance_thresh.  We can removed the parameters when initializing the task at line 497.

Reply:
As to max_steps,  it should also be removed here. We have the gym parameter max_episode_steps so don't bother with this parameter. And most time we should use max_episode_steps, so that the final return of the last step can be correctly handled:
If max_episode_steps is reached and the env is reset, the potential discounted future reward will be take into account when update the value network. Most task should be like this. 
If max_steps is reached with 'done' and the env is reset, the algorithm don't know env reaches time limits and treat this the same as done. All potential discounted reward in the future will be considered as zero. For some task like goal task, that also could work. but using max_episode_steps is better.
If max_steps is reached without 'done' and the env is reset, the algorithm part don't know env is reset, the initial state of next episode will be considered as the next state of the last step. This should never happen.

You can configure the max_episode_steps like this: suite_socialbot.load.max_episode_steps=300

Reply:
Good suggestions.  All done.